geode-management/src/main/java/org/apache/geode/management/configuration/Region.java

```diff
@@ -127,25 +134,75 @@ public void setDiskStoreName(String diskStoreName) {
     this.diskStoreName = diskStoreName;
   }
 
-  /**
-   * two regions are equal if name and type are equal.
-   */
-  @Override
-  public boolean equals(Object that) {
-    if (this == that) {
-      return true;
+  public List<Expiration> getExpirations() {
+    return expirations;
+  }
+
+  public void setExpirations(List<Expiration> expirations) {
+    if (expirations == null) {
+      this.expirations = null;
+      return;
     }
-    if (that == null || getClass() != that.getClass()) {
-      return false;
+    this.expirations = expirations.stream().filter(Objects::nonNull).collect(Collectors.toList());
+  }
+
+  public void addExpiry(ExpirationType type, Integer timeout, ExpirationAction action) {
+    if (expirations == null) {
+      expirations = new ArrayList<>();
     }
-    Region config = (Region) that;
-    return Objects.equals(getName(), config.getName()) &&
-        Objects.equals(getType(), config.getType());
+    expirations.add(new Expiration(type, timeout, action));
   }
 
+  public enum ExpirationType {
+    ENTRY_TIME_TO_LIVE,
+    ENTRY_IDLE_TIME,
+    UNSUPPORTED
```

Comment:
can you add an expiration type more than once?
what does it mean to add an expiration type of UNSUPPORTED?  Should we have javadoc comments on each of these enum values?

Reply:
Validator will catch adding an ExpirationType more than once I believe.